optuna/visualization/_pareto_front.py

```diff
@@ -74,196 +91,142 @@ def objective(trial):
-                "higher than 1."
+                f"higher than {n_dim - 1}."
             )
         if min(axis_order) < 0:
             raise ValueError(
                 f"Given `axis_order` {axis_order} contains invalid index {min(axis_order)} "
                 "lower than 0."
             )
 
-    data = [
-        go.Scatter(
-            x=[t.values[axis_order[0]] for t in trials[n_best_trials:]],
-            y=[t.values[axis_order[1]] for t in trials[n_best_trials:]],
-            text=[_make_hovertext(t) for t in trials[n_best_trials:]],
-            mode="markers",
-            hovertemplate="%{text}<extra>Trial</extra>",
-            marker={
-                "line": {"width": 0.5, "color": "Grey"},
-                "color": [t.number for t in trials[n_best_trials:]],
-                "colorscale": "Blues",
-                "colorbar": {
-                    "title": "#Trials",
-                },
-            },
-            showlegend=False,
-        ),
-        go.Scatter(
-            x=[t.values[axis_order[0]] for t in trials[:n_best_trials]],
-            y=[t.values[axis_order[1]] for t in trials[:n_best_trials]],
-            text=[_make_hovertext(t) for t in trials[:n_best_trials]],
-            mode="markers",
-            hovertemplate="%{text}<extra>Best Trial</extra>",
-            marker={
-                "line": {"width": 0.5, "color": "Grey"},
-                "color": [t.number for t in trials[:n_best_trials]],
-                "colorscale": "Reds",
-                "colorbar": {
-                    "title": "#Best trials",
-                    "x": 1.1 if include_dominated_trials else 1,
-                    "xpad": 40,
-                },
+    def _make_scatter_object(
+        trials: List[FrozenTrial],
+        hovertemplate: str,
+        name: Optional[str] = None,
+        color: Optional[str] = None,
+        dominated_samples: bool = False,
+    ) -> Union["go.Scatter", "go.Scatter3d"]:
+        return _make_scatter_object_base(
+            n_dim,
+            trials,
+            axis_order,  # type: ignore
+            include_dominated_trials,
+            hovertemplate=hovertemplate,
+            name=name,
+            color=color,
+            dominated_samples=dominated_samples,
+        )
+
+    if constraints_func is None:
+        data = [
+            _make_scatter_object(
+                non_best_trials,
+                hovertemplate="%{text}<extra>Trial</extra>",
+                dominated_samples=True,
+            ),
+            _make_scatter_object(
+                best_trials,
+                hovertemplate="%{text}<extra>Best Trial</extra>",
+                dominated_samples=False,
+            ),
+        ]
+    else:
+        data = [
+            _make_scatter_object(
+                infeasible_trials,
+                hovertemplate="%{text}<extra>Infeasible</extra>",
+                name="infeasible",
```

Comment:
As can be seen in the [`plot_optimization_history`](https://optuna.readthedocs.io/en/stable/reference/visualization/generated/optuna.visualization.plot_optimization_history.html#optuna.visualization.plot_optimization_history) document, the names usually start with capital letters in Optuna's plots.
```suggestion
                name="Infeasible",
```